From: Doug Ledford <dledford@redhat.com>
Date: Tue, 16 Sep 2008 16:56:10 -0400
Subject: [openib] add an enum for future RDS support
Message-id: 1221598570.1927.507.camel@firewall.xsintricity.com
O-Subject: [Patch RHEL5.3] Add an enum from OFED
Bugzilla: 462551
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

There has been some talk of possibly trying to enable the RDS protocol
(currently part of the OFED stuff, but I exclude it since it hasn't gone
to Linus' kernel yet) sometime prior to rhel5.4.  I've looked over all
the patches necessary to enable support *for* RDS without actually
putting RDS into our kernel.  There is only one and I'm attaching it
here.

There currently is no bugzilla for this, it has grown out of a request
from Oracle.  In particular, Oracle is planning to make RDS a
requirement in the future for RAC cluster configurations and
certifications.  However, RDS still isn't ready for inclusion IMO due to
the fact that it hasn't been submitted to Linus' kernel yet and because
there is talk about changing it from a net protocol that works over
TCP/IP or IB to just an IB ULP.  So, we didn't want to put it in 5.3,
but waiting until 5.4 will leave us uncertified for a long time.

All sorts of higher up management types have been in on these talks and
since this is *SO* minor (and already in OFED 1.4, so no danger of
allocating an enum that conflicts with upstream later), it was deemed
best to get this into 5.3 if for no other reason than to keep our
options open on how/when to support RDS in the future.

--
Doug Ledford <dledford@redhat.com>
              GPG KeyID: CFBFF194
              http://people.redhat.com/dledford

Infiniband specific RPMs available at
              http://people.redhat.com/dledford/Infiniband

From 86cba4b2dfc6600d3b8af298047c01e8ac333810 Mon Sep 17 00:00:00 2001
From: Andy Grover <andy.grover@oracle.com>
Date: Thu, 4 Sep 2008 14:26:26 -0700
Subject: [PATCH] RDS: Fix compilation error

Backport the addition of an enum entry that is used by RDS

Signed-off-by: Andy Grover <andy.grover@oracle.com>

diff --git a/include/rdma/rdma_cm.h b/include/rdma/rdma_cm.h
index 010f876..57f7d07 100644
--- a/include/rdma/rdma_cm.h
+++ b/include/rdma/rdma_cm.h
@@ -53,7 +53,8 @@ enum rdma_cm_event_type {
 	RDMA_CM_EVENT_DISCONNECTED,
 	RDMA_CM_EVENT_DEVICE_REMOVAL,
 	RDMA_CM_EVENT_MULTICAST_JOIN,
-	RDMA_CM_EVENT_MULTICAST_ERROR
+	RDMA_CM_EVENT_MULTICAST_ERROR,
+	RDMA_CM_EVENT_ADDR_CHANGE,
 };
 
 enum rdma_port_space {
